android/BOINC/app/build.gradle

```diff
@@ -1,5 +1,45 @@
+plugins {
+    id 'org.ajoberstar.grgit' version '1.7.1'
+    id 'com.palantir.git-version' version '0.7.1'
+}
+
 apply plugin: 'com.android.application'
 
+import org.ajoberstar.grgit.Grgit
+
+// Use commit date as version code
+def buildVersionCode() {
+    def repo = Grgit.open()
+    def head = repo.head()
+
+    // Sanity check across git plugins
+    assert head.getAbbreviatedId(10) == versionDetails().gitHash : "Internal error: SHA1 mismatch!"
+
+    return head.time
+}
+
+// Derive version name from release tag and add commit SHA1
+def buildVersionName() {
+    def pattern = /client_release\/\d+\.\d+\/(?<major>\d+)\.(?<minor>\d+)\.(?<revision>\d+)(?<suffix>[-_].*)/
+    def message = 'Invalid release tag encountered:'
+
+    def tag = versionDetails().lastTag
```

Comment:
Yes, if there is a proper tag and no offset display the version number like `7.6.33 (6f6c8b7eb5)`. If there is an offset or the tag pattern does not match, display `DEVELOPMENT (981b0409d3)`